crates/llvm-context/src/polkavm/context/pointer.rs

```diff
@@ -76,6 +76,35 @@ impl<'ctx> Pointer<'ctx> {
         Self::new(r#type, address_space, value)
     }
 
+    /// Creates a new pointer with the specified `offset` for storage operations.
+    pub fn new_with_store_offset<D, T>(
```

Comment:
The right place for this would be `fn build_storage_pointer_argument_store` on context, similar to the other helpers.